pkg/crc/config/viper_config.go

```diff
@@ -86,8 +104,8 @@ func EnsureConfigFileExists() error {
 	return nil
 }
 
-// InitViper initializes viper
-func InitViper() error {
+// NewConfig initializes viper
+func NewConfig() (*ViperConfig, error) {
 	v := viper.New()
 	v.SetConfigFile(constants.ConfigPath)
```

Comment:
This could (should?) be a singleton as we don't want to instantiate multiple `ViperConfig` writing to the same config file. If it's a singleton, maybe we don't need to add it as an arg to some methods, but just use `config.GetViperConfig()` instead? (just starting a conversation, not saying it has to be done this way)

Reply:
If it is a singleton, it will always use the same file and this is what I want to avoid.
Today, if I run the unit test, it will change my crc configuration. I don't want unit tests to change anything on my system.

Reply:
> If it is a singleton, it will always use the same file and this is what I want to avoid.

But `NewConfig()` as it is now will always use the same file because of `v.SetConfigFile(constants.ConfigPath)`. I guess this is work in progress :)